sim/dsl.go

```diff
@@ -366,6 +366,37 @@ func (s Simulator) execute() {
 			p := s.getPuppet(name)
 			p.caps = caps
 			instr.TestSuccess()
+		case "reset":
+			name := s.getInstructionArg(1)
+			langImpl := s.getInstructionArg(2)
+			if _, ok := s.implementations[langImpl]; !ok {
+				err := errors.New(fmt.Sprintf("no such language implementation passed to simulator on startup (%s)", langImpl))
+				s.Abort(err)
+				return
+			}
+			p := s.getPuppet(name)
+			// puppet directory was empty => it was never started
+			if p.directory == "" {
+				instr.TestSuccess()
+				taplog(fmt.Sprintf("there was no execution folder to reset for %s", p.name))
+				continue
+			}
+			subfolder := fmt.Sprintf("%s-%s", langImpl, name)
+			fullpath := filepath.Join(s.puppetDir, subfolder)
+
+			absdir, err := filepath.Abs(fullpath)
+			if err != nil {
+				s.Abort(fmt.Errorf("%s errored during reset (%w)", p.name, err))
+				return
+			}
+			// remove the created puppet dir, thus resetting its starting state
+			err = os.RemoveAll(absdir)
+			if err != nil {
+				s.Abort(fmt.Errorf("%s errored during reset (%w)", p.name, err))
+				return
+			}
+			instr.TestSuccess()
+			taplog(fmt.Sprintf("removed %s", p.name, subfolder))
```

Comment:
Good spotting @mycognosist! https://github.com/ssb-ngi-pointer/netsim/pull/35#issuecomment-933420185

This fmt.Sprintf has two arguments but just one formatting instruction :)

Reply:
ah woopsidoopsie, think i changed what i had the second before committing lol

(decide not to leak the folder in the output)

Reply:
fixed in dac4630, thanks!